run.go

```diff
@@ -168,6 +170,27 @@ func RunGame(game Game) error {
 	return nil
 }
 
+// RunOnMainThread calls the given f on the main thread, and blocks until f returns.
+//
+// This method panics if f panics.
+func RunOnMainThread(f func()) {
+	var err error
+	if graphicscommand.RunOnMainThread(func() {
+		// If f panics, capture the panic error and propagates to the
+		// calling goroutine.
+		defer func() {
+			if r := recover(); r != nil {
+				err = fmt.Errorf("%v", r)
```

Comment:
Hmm, this loses the information about the stack trace.

IMO we don't have to `recover` a panic.

Reply:
I thought about this, but I saw this one:
https://github.com/hajimehoshi/ebiten/pull/1927/files#diff-136576161d4f7b4c21043aa608a2b7e7d8acb542d5c7db83d17f57f2a6a6aa2aL617

If the given f panics on the main thread, then the main thread can block forever because the done channel will never be signaled.

Maybe we just send the done in a deferred call.

Reply:
This panic should be moved out of the closure.